testplan/common/utils/logger.py

```diff
@@ -82,7 +82,12 @@ def driver_info(self, msg, *args, **kwargs):
 
     def log_test_status(self, name, passed, indent=0, level=TEST_INFO):
         """Shortcut to log a pass/fail status for a test."""
-        pass_label = Color.green('Pass') if passed else Color.red('Fail')
+        if passed is True:
+            pass_label = Color.green('Pass')
+        elif passed is False:
+            pass_label = Color.red('Fail')
+        else:
```

Comment:
This is weird - so `passed` is no longer a simple boolean - what else could it be? I think that this method should be changed to accept the status string instead of a `passed` boolean, in order to correctly handle XFAIL and possible future test states (SKIP, XPASS etc.)